### PR TITLE
dep: replace `colors` with `chalk`

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

The pull-request updates the problematic [colors.js](https://github.com/Marak/colors.js) dependency with [chalk](https://github.com/chalk/chalk). In addition, the migration guide was also updated to include a note about the possibility of using resolutions in order not to pull the problematic dependency version (which may be pulled transitively through `electron-rebuild` for example).

#### How to test

1. confirm that the build is successful
2. confirm that starting the example applications is successful
3. confirm that the downloading plugins script works, and that output is correctly colored (success, skip, errors)

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Review by msujew — APPROVED** [review]
I can confirm that the application starts correctly and that the `download:plugins` command works as expected (with colors and everything).

With the changes in place, no direct references to the `colors` package exist anymore in the repo 👍

File: doc/Migration.md [hunk 1]
```diff
@@ -7,6 +7,18 @@ Please see the latest version (`master`) for the most up-to-date information. Pl
...
 
+### General
+
+Due to a [colors.js](https://github.com/Marak/colors.js) issue, a [resolution](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) may be necessary for your application in order to workaround the problem:
+
+For example:
+
+```json
+"resolutions": {
+    "**/colors": "1.4.0"
```
Comment (marcdumais-work) [inline doc/Migration.md:18]: Maybe we can be more generic and not assume the version needed, so long as it's not the newest one? e.g.:
`"**/colors": "<=1.4.0"`

Comment (vince-fugnitto) [reply doc/Migration.md]: I confirmed that `<=1.4.0` works correctly when building a downstream application, and adjusted the migration documentation.

**Review by marcdumais-work — APPROVED** [review]
LGTM, thanks Vince!

**Pull request merged**